### ISSUE TITLE
Toggle and ToggleSmall aria-label is better placed on the checkbox, not the label element

### ISSUE THREAD

**Issue by FrivalszkyP** [carbon-design-system/carbon#6567]
## Environment

> MacOS

> Chrome

> VoiceOver

## Detailed description

VoiceOver on Chrome does not announce the aria-label of the Toggle element because the attribute is on the label element, not the input itself.

> What did you expect to happen?

I expected the text of the aria-label to be announced by VoiceOver upon focusing on the toggle element.

> What happened instead?

The text was not announced, only the "tick box" and its on/off state. 

> What WCAG 2.1 checkpoint does the issue violate?

Looking at the "accessible name computation" specification, it seems to me that the current setup, having an aria-label on a label, is not in the specification. The aria-label attribute would be better suited on the input element itself.

## Steps to reproduce the issue

1. Turn on VoiceOver for Chrome
2. Go to [Storybook](https://react.carbondesignsystem.com/?path=/story/togglesmall--toggled)
2. Navigate to the Toggle element

## Additional information

- It's important to note that the label is announced in Safari, which is probably the most used browser with VoiceOver
- I was unable to test this with NVDA/JAWS at the moment
- But as per specification, I thing the aria attribute should go onto the input element anyways. 
- I tested it by moving the attribute using the Developer Tools, and it worked both with Safari and Chrome

Let me know if you found my reasoning solid, and I'll go ahead and create a PR!

**Comment by emyarod** [comment]
I am able to get VO to read out the aria-label on toggles at least in the storybook example, but I will await the feedback from our a11y experts about moving the label attribute to the input!

**Comment by tw15egan** [comment]
@dakahn any thoughts on this one?

**Comment by dakahn** [comment]
We build for VoiceOver used in Safari as, like you pointed out, that is the ideal and most used browser pairing generally

`aria-label` is unneeded. The toggle input is explicitly labeled by being wrapped in a `label` element. I've opened a PR to update the docs and demonstrate proper use of this component with a visual label [at this link](https://github.com/carbon-design-system/carbon/pull/6770)

That said, this component has problems. First and foremost [WCAG Success Criterion 3.3.2](https://www.w3.org/TR/WCAG21/#labels-or-instructions) requires that form inputs have accessible labeling for both sighted users _and_ users of assistive technologies. For a deeper dive [see the section **Intent** in this Labeling and Instructions guide](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html), 

Currently we allow users to hide the label (perhaps to set it themselves and hopefully not to "save space"). But with JAWS 2020 and an `aria-label` attribute set on the label parent element -- no label is read 😢.

I think the `label` element should become a _sibling_ to the `<Toggle>` instead of a parent like it currently is with it's `id` set as an `aria-labelledby` attribute on the input. 

If users don't want _our_ visual label on the element directly we should just drop `<label>` altogether and allow them to set their own label via `aria-labelledby` on the input itself.

**Comment by tw15egan** [comment]
Does not to be an issue anymore when testing in VoiceOver in Chrome in the V11 storybook 👍🏻
